### PR TITLE
WT-4643 Stable timestamp can move past durable timestamp of active txns

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: src/txn/txn_timestamp.c [hunk 1]
```diff
@@ -637,7 +637,7 @@ __wt_txn_set_commit_timestamp(
 			    ts_string[0], ts_string[1]);
 		}
 
-		if (has_stable_ts && commit_ts < stable_ts) {
+		if (has_stable_ts && commit_ts <= stable_ts) {
```
Comment (bvpvamsikrishna) [inline src/txn/txn_timestamp.c]: update the error msg to reflect the condition.
This condition change is needed for durable timestamp check as well in set_durable_timestamp

**Comment by luke-pearson** [comment]
retest this please

**Comment by luke-pearson** [comment]
retest this please

**Comment by luke-pearson** [comment]
retest this please

**Comment by luke-pearson** [comment]
retest this please

**Comment by luke-pearson** [comment]
retest this please

File: src/txn/txn_timestamp.c [hunk 2]
```diff
@@ -688,8 +688,8 @@ __wt_txn_set_commit_timestamp(
 
 	if (!F_ISSET(txn, WT_TXN_HAS_TS_PREPARE)) {
 		/*
-		 * For a non-prepared transactions the commit timestamp should
-		 * not be less than the stable timestamp.
+		 * For non-prepared transactions the commit timestamp should
+		 * not be less than the oldest timestamp.
 		 */
 		if (has_oldest_ts && commit_ts < oldest_ts)
```
Comment (bvpvamsikrishna) [inline src/txn/txn_timestamp.c]: the condition should be `commit_ts <= oldest_ts`

File: src/txn/txn_timestamp.c [hunk 3]
```diff
@@ -784,18 +788,22 @@ __wt_txn_set_durable_timestamp(
 		stable_ts = txn_global->stable_timestamp;
 
 	/*
-	 * For a non-prepared transactions the commit timestamp should
-	 * not be less than the stable timestamp.
+	 * The durable timestamp should not be less than the oldest timestamp.
 	 */
 	if (has_oldest_ts && durable_ts < oldest_ts)
```
Comment (bvpvamsikrishna) [inline src/txn/txn_timestamp.c]: the condition should be `durable-ts <= oldest-ts`

File: src/txn/txn_timestamp.c (hunk 2 above)
Comment (luke-pearson) [reply src/txn/txn_timestamp.c]: As discussed we are going to keep it as just less than the oldest timestamp as the behavior would conflict with the rounding up to oldest timestamp logic.

**Pull request merged**